registry/client/src/settings/Edit.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {
+    Edit,
+    SimpleForm,
+    TextInput,
+    RadioButtonGroupInput,
+    BooleanInput,
+    PasswordInput,
+    Toolbar,
+    SaveButton,
+} from 'react-admin';
+
+import * as validators from '../validators';
+import {types} from './dataTransform';
+
+const Title = (props) => {
+    return (<span>{props.record ? `Setting "${props.record.key}"` : ''}</span>);
+};
+
+const MyToolbar = (props) => {
+    return (
+        <Toolbar {...props}>
+            <SaveButton />
+        </Toolbar>
+    );
+};
+
+const Input = (props) => {
+    switch(props.record.meta.type) {
+        case types.url: {
+            return (<TextInput source="value" validate={validators.url} />);
+        }
+        case types.enum: {
+            return (<RadioButtonGroupInput row={false} source="value" choices={props.record.meta.choices} />);
+        }
+        case types.boolean: {
+            return (<BooleanInput source="value" />);
+        }
+        case types.password: {
+            return (<PasswordInput source="value" />);
+        }
+        default: {
+            return (<TextInput source="value" />);
+        }
+    }
+};
+
+const MyEdit = (props) => {
+    return (
+        <Edit title={<Title />} undoable={false} {...props}>
```

Comment:
Could you attach a screenshot of this page?

Reply:
<img width="1440" alt="Screenshot 2020-10-20 at 12 50 10" src="https://user-images.githubusercontent.com/57893569/96570267-d28fbe80-12d2-11eb-9b2d-7907364edce6.png">